src/period-calculation/getYearlyPeriods.ts

```diff
@@ -0,0 +1,167 @@
+  FixedPeriod,
+  GeneratedPeriodsFunc,
+  PeriodIdentifier,
+} from "./fixed-periods";
+import { monthNumbers } from "./getMonthlyPeriods";
+
+export const getYearlyPeriods: GeneratedPeriodsFunc = ({
+  year,
+  calendar,
+  periodType,
+  yearsCount = 10,
+  locale,
+}) => {
+  const month = getMonth(periodType);
+  const currentYear = Temporal.PlainDate.from({
+    year,
+    month,
+    day: calendar.toString() === "nepali" ? 14 : 1,
+    calendar,
+  });
+
+  const years: FixedPeriod[] = [];
+
+  for (let i = 0; i < yearsCount; i++) {
+    const dateToAdd = currentYear.subtract({ years: i });
+    const value = buildValue({
+      periodType,
+      year: dateToAdd.year,
+      month,
+    });
+    const year: FixedPeriod = {
+      id: value,
+      iso: value,
+      name: buildLabel(periodType, dateToAdd, {
+        locale: locale || "en",
+        calendar,
+      }),
+      ...buildStartAndEndDates(dateToAdd),
+    };
+    if (isFinancialYear(periodType)) {
+      delete year.iso;
+    }
+    years.push(year);
+  }
+  return years;
+};
+
+const buildStartAndEndDates = (date: Temporal.PlainDate) => {
+  const endDate = Temporal.PlainDate.from({
+    year: date.year + 1,
+    month: date.month,
+    day: 1,
+    calendar: date.calendar,
+  }).subtract({ days: 1 });
+  return {
+    startDate: formatYyyyMmDD(date, "startOfMonth"),
+    endDate: formatYyyyMmDD(endDate, "endOfMonth"),
+  };
+};
+
+const getMonth = (periodType: string) => {
+  const yearType = periodType?.replace("FY", "").toUpperCase();
+  const monthInfo = monthNumbers[yearType as keyof typeof monthNumbers];
+
+  return monthInfo?.value || 1;
+};
+const buildLabel = (
+  periodType: PeriodIdentifier,
+  currentYearDate: Temporal.PlainDate,
+  options: { locale: string; calendar: SupportedCalendar }
+) => {
+  if (periodType === "YEARLY") {
+    return currentYearDate.year.toString();
+  }
+
+  if (isCustomCalendar(options.calendar)) {
+    return buildLabelForCustomCalendar(currentYearDate, options);
+  }
+  const format = {
+    month: "long" as const,
+    year: "numeric" as const,
+    calendar: options.calendar,
+  };
+  const fromYear = localisationHelpers.localiseMonth(
+    currentYearDate,
+    options,
+    format
+  );
+
+  const toYear = localisationHelpers.localiseMonth(
+    currentYearDate.add({ months: currentYearDate.monthsInYear - 1 }),
+    options,
+    format
+  );
+
+  let result = `${fromYear} - ${toYear}`;
+
+  // needed for Ethiopic calendar
+  result = result.replace(/ERA\d+\s*/g, "").trim();
+
+  return result;
+};
+
+const buildLabelForCustomCalendar = (
+  date: Temporal.PlainDate,
+  options: { locale: string; calendar: SupportedCalendar }
+) => {
+  const localiseMonth = (dateToDisplay: Temporal.PlainDate) =>
+    `${localisationHelpers.localiseMonth(dateToDisplay, options, {})} ${
+      dateToDisplay.year
+    }`;
+
+  const nextYearDate = Temporal.PlainDate.from({
+    year: date.year + 1,
+    month: (date.month - 1) % date.monthsInYear,
```

Comment:
I don't think I follow this. Ia this to handle calendars with an "extra" month or something? Won't date.month -1 be less than the date.monthsInYear?